bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/BRC1HUartProcessor.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.concurrent.PriorityBlockingQueue;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.ResponseListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+@NonNullByDefault
+public class BRC1HUartProcessor {
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * In the unlikely event of messages arrive in wrong order, this comparator will sort the queue
+     */
+    private Comparator<byte[]> chunkSorter = (byte[] m1, byte[] m2) -> m1[0] - m2[0];
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private final Logger logger = LoggerFactory.getLogger(BRC1HUartProcessor.class);
+
+    private PriorityBlockingQueue<byte[]> uartMessages = new PriorityBlockingQueue<>(10, chunkSorter);
```

Comment:
It should be noted that this class iterator elements in order, it only guarantees that polling from it will return the smallest element. I think what you might want to use instead is a ConcurrentSkipListSet which does has iteration ordering guarantees.

Reply:
Thanks for spotting this - I actually did the unit test with pull - and refactored the code then iterating over. Will fix.